pages/api/loadNew.ts

```diff
@@ -0,0 +1,102 @@
+import * as fs from "fs";
+import path from "path";
+import { NextApiRequest, NextApiResponse} from 'next'
+
+import { getJsonStructure } from "evergreen-org-crawler/src/index"
+import config from "../../config.json"
+
+// Cache files are stored inside ./next folder
+const CachePath = path.resolve("./dynamicCache.json")
+const EmptyCachePath = path.resolve("./defaultDynamicCache.json")
+
+//TODO: Move to config file
+const timeUntilRefresh = 5 * 60 * 1000 // 5 minutes in milliseconds
+
+let waitingPromise: Promise<void> | null = null
+let promiseResolve: any = null
+
+function getProperty<T, K extends keyof T>(o: T, propertyName: K): T[K] {
+    return o[propertyName];
+}
+
+async function createData(request: "npm" | "PyPI" | "RubyGems" | null = null){
+	const requestToAPI = {
+		npm: "NPM",
+		PyPI: "PYPI",
+		RubyGems: "RUBYGEMS"
+	}
+
+	let api = null
+	if(request != null){
+		api = [getProperty(requestToAPI, request)]
+	}
+
+	const accessToken = process.env.NEXT_PUBLIC_GITHUB_TOKEN!
+	return getJsonStructure(accessToken, config, api)
+}
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
```

Comment:
I see very little comments. The file is called loadNew.ts, but load new what? how? from where?

Also, we should define some typings for what the handler returns, so that other parts of the code can just use that type when they call the API.